linalg/hypre.cpp

```diff
@@ -926,6 +933,11 @@ void HypreParMatrix::GetOffd(SparseMatrix &offd, HYPRE_Int* &cmap) const
    cmap = A->col_map_offd;
 }
 
+void HypreParMatrix::GetProcRows(SparseMatrix &colCSRMat)
+{
+   MakeWrapper(hypre_MergeDiagAndOffd(A), colCSRMat);
```

Comment:
This leaks the `hypre_CSRMatrix` returned by `hypre_MergeDiagAndOffd`. However, since `colCSRMat` just wraps the data of that matrix, we can not simply destroy that `hypre_CSRMatrix`.

We may need create a temporary `SparseMatrix` wrapper, copy it to copy the data to the result `colCSRMat` and then destroy the intermediate `hypre_CSRMatrix` and the temporary `SparseMatrix` wrapper.

Simply transferring the ownership from `hypre_CSRMatrix` to the result `SparseMatrix` is not correct (at least at the moment) because of how hypre allocates memory and how `SparseMatrix` deletes memory -- the former uses standard c library functions while the latter uses standard C++ `delete`. If we add a new `mfem::MemoryType` for allocation/de-allocation with standard c library functions, then we won't need to copy the data. @camierjs, this is probably worth considering as an addition to the memory manager.

Reply:
For now I went with the first option. The `MemoryType` idea could also be interesting to pursue.